test/assignment4/Assignment4Test.java

```diff
@@ -0,0 +1,77 @@
+package assignment4;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import javax.imageio.ImageIO;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class Assignment4Test {

```

Comment:
Clever use of images!

More idiomatic way to handle these though would be via [parameterized tests](https://github.com/junit-team/junit4/wiki/parameterized-tests).